### ISSUE TITLE
Use directive ignored for prefixes

### ISSUE THREAD

**Issue by marekdedic** [humbug/php-scoper#456]
## Bug report

| Question    | Answer
| ------------| ---------------
| PHP-Scoper version | 0.14.0
| PHP version | 7.4.15
| Platform with version | Debian testing
| Github Repo | Not public

Hi,
I'm trying to use php-scoper on the Google SDK and I am getting prefixes for classes that have already been brought into scope with the `use` directive. I tried shortening it a bit, but here's my example:

Source file:
```php
<?php
namespace Google\AccessToken;

use Google\Auth\HttpHandler\HttpHandlerFactory;
use Google\Client;
use GuzzleHttp\ClientInterface;
use GuzzleHttp\Psr7;
use GuzzleHttp\Psr7\Request;

/**
 * Wrapper around Google Access Tokens which provides convenience functions
 *
 */
class Revoke
{
  /**
   * @var ClientInterface The http client
   */
  private $http;

  /**
   * Instantiates the class, but does not initiate the login flow, leaving it
   * to the discretion of the caller.
   */
  public function __construct(ClientInterface $http = null)
  {
    $this->http = $http;
  }
}
```

Output:
```php
<?php
namespace Sgdg\Vendor\Google\AccessToken;

use Sgdg\Vendor\Google\Auth\HttpHandler\HttpHandlerFactory;
use Sgdg\Vendor\Google\Client;
use Sgdg\Vendor\GuzzleHttp\ClientInterface;
use Sgdg\Vendor\GuzzleHttp\Psr7;
use Sgdg\Vendor\GuzzleHttp\Psr7\Request;
/**
 * Wrapper around Google Access Tokens which provides convenience functions
 *
 */
class Revoke
{
    /**
     * @var ClientInterface The http client
     */
    private $http;
    /**
     * Instantiates the class, but does not initiate the login flow, leaving it
     * to the discretion of the caller.
     */
    public function __construct(\Sgdg\Vendor\GuzzleHttp\ClientInterface $http = null)
    {
        $this->http = $http;
    }
}
```

As you can see, in the constructor, the `ClientInterface` class is prefixed, but AFAIK that is not necessary as it has already been brought into scope with the `use` directive.

**Comment by theofidry** [comment]
I'm really not sure this should be changed to be honest. Trying to fix it only adds more clutter and makes the prefixing slower for very little benefit (only a CS improvement).

I would also note that the CS of the prefixed code is very very minimal (it's basically the PHP-Parser printer) and any CS change could be done with a more appropriate tool for it (like PHP-CS-Fixer for example).

**Comment by marekdedic** [comment]
Hi,
I've tried to make the necessary modifications in #457 - it's not perfect, but it should cover like 95% of the cases.

- Clutter: It essentially boils down to one if statement. IMO insignificant in the context of the whole library.
- Time: I "benchmarked" it on the `vendor` folder of one of my projects (~6 000 files). The time decreased slightly 23.57s -> 21.49s (within the margin of error I'd say)
- Code size: Decreased by 2.5% - I expected more TBH, but I'll take it :smile: 
- Code readability: Much better, it's much closer to the original, now php-scoper basically just changes the `namespace` and `use` statements for most files.

So, in summary, I think that you could use a CS fixer to do much of the same, but with this change, you get faster runtime, smaller and better readable and debuggable output for a miniscule price...

If I've "sold" it to you sufficiently, please do take a look at the PR :smile: I'm still working on some edge cases and will mark it ready after they're solved.
